packages/firestore/src/platform/node/grpc_connection.ts

```diff
@@ -51,6 +55,7 @@ function createMetadata(databasePath: string, token: Token | null): Metadata {
       }
     }
   }
+  metadata.set('x-firebase-gmpid', appId);
```

Comment:
Question: the plumbing accounts for the possibility of `appId` being empty. What do you think about not setting the header at all if `appId` is empty rather than setting an "empty" header?

Reply:
I looked at the RTDB code and it uses an empty app ID and sends the header. I wanted to match that behavior.